qa/tasks/cephfs/test_multifs.py

```diff
@@ -0,0 +1,397 @@
+        4. Test read and write on both FS.
+        5. Teardown and set up again.
+        6. Repeat 1-5 for all possible combinations produced by read/write
+           modes, fsids and paths.
+        """
+        testcases = (
+            # mds cap without path and with rw permission
+            {
+                'mdscap': 'allow rw fsid={cephfs1_id}, allow rw ' + \
+                          'fsid={cephfs2_id}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': None},
+            # mds cap without path and with r permission
+            {
+                'mdscap': 'allow r fsid={cephfs1_id}, allow r ' + \
+                          'fsid={cephfs2_id}',
+                'osdcap': 'allow r tag cephfs data={cephfs1_name}, allow r tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': None},
+
+            # mds cap with path and with rw permission
+            {
+                'mdscap': 'allow rw fsid={cephfs1_id} path=/{cephfs_mntpt}, allow rw ' + \
+                          'fsid={cephfs2_id} path=/{cephfs_mntpt}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': 'dir1'},
+
+            # mds cap with path and with r permission
+            {
+                'mdscap': 'allow r fsid={cephfs1_id} path=/{cephfs_mntpt}, allow r ' + \
+                          'fsid={cephfs2_id} path=/{cephfs_mntpt}',
+                'osdcap': 'allow rw tag cephfs data={cephfs1_name}, allow rw tag cephfs ' + \
+                          'data={cephfs2_name}',
+                'cephfs_mntpt': 'dir1'},
+
+            # misbehaves
+            #{'mdscap': 'allow rw *', 'osdcap': 'allow rw tag cephfs data=*',
+            # 'cephfs_mntpt': '/'},
+            #{'mdscap': 'allow r *', 'osdcap': 'allow r tag cephfs data=*',
+            # 'cephfs_mntpt': '/'},
+        )
+
+        testcases_passed_count = 0
+        for testcase in testcases:
+            try:
+                self.run_test_auth_for_multiple_fss(**testcase)
+            except Exception as e:
+                print(e)
+                print('=' * 79)
+                print(str(testcases_passed_count) + ' test cases passed '
+                      'successfully.')
+                print('testcase details -')
+                print('mdscap = ' + testcase['mdscap'])
+                print('osdcap = ' + testcase['osdcap'])
+                if isinstance(testcase['cephfs_mntpt'], str):
+                    print('cephfs_mntpt = ' + testcase['cephfs_mntpt'])
+                else:
+                    print('cephfs_mntpt is {}'.format(testcase['cephfs_mntpt']))
+                print('=' * 79)
+                raise
+
+            self.run_test_auth_for_multiple_fss_resetup()
```

Comment:
Since setUp() runs before the test method, I don't see a way ATM to modify `setUp()`'s behaviour based on the test method.

I agree that right now it's much more complex than what a simple average test should be. I've tried to simplify it a bit more in last update. I'll try to simplify it further.